### PR TITLE
initial commit of docker desktop extension

### PR DESCRIPTION
**Pull request by danleyb2**
### Summary by CodeRabbit

- New Feature: Introduced a Docker Desktop extension for the Plate Recognizer Installer, featuring a React frontend and Python backend.
- New Feature: Added a `/verify-token` endpoint in the Flask application for token verification.
- Chore: Set up initial project structure with Dockerfile, Makefile, and ESLint configuration.
- Documentation: Provided instructions for local development, building, and deploying the extension.
- Chore: Configured `.gitignore` to exclude unnecessary files from version control.

### CODE REVIEW

File: docker/dd-extension/logo.svg:Zone.Identifier [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[ZoneTransfer]
```
Comment (Pefington) [inline docker/dd-extension/logo.svg:Zone.Identifier:1]: You can remove this file, it's windows metadata dumped when copying from NTFS to an other filesystem.

File: docker/dd-extension/ui/src/components/Snapshot.tsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    openBrowserUrl(ddClient, e.target.href);
+  };
+  return (
+    <Form onSubmit={handleSubmit}>
+      <Loader isLoading={isLoading} />
+
+      <Form.Group as={Row} className="mb-3" controlId="snapshotToken">
+        <Form.Label column sm={4}>
+          Please enter your Plate Recognizer{" "}
+          <a href="https://app.platerecognizer.com/accounts/plan/#sdk/?utm_source=dd-extension&utm_medium=app" onClick={handleLinkClick}>
```
Comment (Pefington) [inline docker/dd-extension/ui/src/components/Snapshot.tsx:138]:
```suggestion
          <a href="https://app.platerecognizer.com/service/snapshot-sdk/" onClick={handleLinkClick}>
```

File: docker/dd-extension/ui/src/components/Snapshot.tsx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+            name="token"
+            onChange={handleInputChange}
+          />
+        </Col>
+      </Form.Group>
+
+      <Form.Group as={Row} className="mb-3" controlId="snapshotLicense">
+        <Form.Label column sm={4}>
+          Please enter your{" "}
+          <a href="https://app.platerecognizer.com/accounts/plan/#sdk/?utm_source=dd-extension&utm_medium=app" onClick={handleLinkClick}>
```
Comment (Pefington) [inline docker/dd-extension/ui/src/components/Snapshot.tsx:157]:
```suggestion
          <a href="https://app.platerecognizer.com/service/snapshot-sdk/" onClick={handleLinkClick}>
```

File: docker/dd-extension/ui/src/components/Snapshot.tsx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+            placeholder="Port"
+            name="port"
+            required
+            defaultValue={8080}
+            onChange={handleInputChange}
+          />
+        </Col>
+      </Form.Group>
+
+      <Form.Group as={Row} className="mb-3" controlId="snapshotDockerImage">
```
Comment (Pefington) [inline docker/dd-extension/ui/src/components/Snapshot.tsx:204]: Could you make two separate dropdowns?
- One for the country
- One for the architecture
Up to date reference: https://guides.platerecognizer.com/docs/snapshot/manual-install/

**Review by Pefington — CHANGES_REQUESTED** [review]
This works, good stuff.

- Some of the URLs need updating (marked).
- For snapshot, we need two dropdowns, one for country, one for architecture (commented).

More globally, I see a lot of small issues with the TSX code, would you mind adding eslint and fixing the errors?
I think it would make the code cleaner and more maintainable:

- cd ..../ui
- `npm init @eslint/config@latest`
  - check syntax and find problems
  - js modules
  - react
  - yes for typescript
  - node
  - yes to install
- Install the VSCode `eslint` extension.

The generated config should look like this:
```js
import globals from "globals";
import pluginJs from "@eslint/js";
import tseslint from "typescript-eslint";
import pluginReact from "eslint-plugin-react";

export default [
  {files: ["**/*.{js,mjs,cjs,ts,jsx,tsx}"]},
  {languageOptions: { globals: globals.browser }},
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  pluginReact.configs.flat.recommended,
];
```

Don't hesitate to ping me if you need!

**Comment by danleyb2** [comment]
@Pefington added eslint however some the lint errors such as `@typescript-eslint/no-explicit-any` are unresolved

**Comment by Pefington** [comment]
> @Pefington added eslint however some the lint errors such as `@typescript-eslint/no-explicit-any` are unresolved

Thank you, I'll check in the morning. 
Could you see if you can resolve some of the security issues the bot is mentioning, I'll have a look at it too.

**Comment by danleyb2** [comment]
@Pefington This bot is autodated and got recently replaced with **coderabbit**. Let's ignore it. I'll try run **coderabbit**

**Review by Pefington — APPROVED** [review]

**Comment by Pefington** [comment]
@danleyb2 Thank you, the changes are good, everything works.
I'll let you decide if you want to implement coderabbit's unresolved suggestions.
I think they are valid, but it's mostly nit.

**Comment by danleyb2** [comment]
@Pefington I'll resolve a number then merge

**Pull request merged**